### DISCUSSION TITLE
How to publish paper items with customed IDs (e.g., no doi, pmd, etc)?

### DISCUSSION THREAD

**Discussion by imnanxu** [greenelab/lab-website-template#270] [Q&A] [answered]
### Checks

- [X] I have searched **[the docs](https://greene-lab.gitbook.io/lab-website-template-docs)**, [existing issues](https://github.com/greenelab/lab-website-template/issues), and [existing discussions](https://github.com/greenelab/lab-website-template/discussions) for answers first.

### Link to your website repo

https://github.com/inspirelab-site/inspirelab-site.github.io/tree/vscode072524

### Version of Lab Website Template you are using

1.2.2

### Description

I have an issue with adding conference papers with no official ids (dot, pmd, etc..). I ended up creating unofficial ids for these items (like ohbm2024i, etc). I realized that on my local setup, vscode+docker can still deploy them on the webpage as expected despite reporting errors in the terminal:
```
Processing source 25 of 37, id: ohbm2024i
    Using Manubot to generate citation## WARNING
Generating csl_item for 'ohbm2024i' failed due to a NotImplementedError:
Manubot does not know how to generate a csl_item for 'ohbm2024i'

            Couldn't parse Manubot response
```
However, on the GitHub web gui (see vscode072524), these errors would prevent the deploy steps so that I couldn't merge to the main. I wonder if there may be a way to get around of this errors?

P.S. I also tried removing the IDs for these items, like just using - Title: "" ..., but the website shows a `[no id info]` at the end of the published items, which looks odd. It would be great if there's a way to keep these costumed id name without reporting errors.

**Comment by vincerubinetti** [comment, marked as answer]
> I realized that on my local setup, vscode+docker can still deploy them on the webpage as expected despite reporting errors in the terminal

When running locally, yes, a critical error in the cite process won't prevent the site preview from building. But the cite process still would not have completed, so any changes you made to your sources/orcid/etc since the last successful run wouldn't be there on your site.

If you want to prevent the cite process from throwing a _critical error_ when Manubot cant cite a source (e.g. when using a non-standard `id`), find `# if Manubot cannot cite source` `except Exception as e:` and change `"ERROR"` to `"WARNING"` and comment out `error = True`. _I wouldn't recommend doing this_. One day you might try to type in a standard `id`, mis-type it, and not realize that you had because nothing failed and you didn't check the logs for a warning.

I _think_ you could also put `continue-on-error: true` under `jobs:` `update-citations:` in `on-pull-request.yml` and `on-push.yml`, so that your site can still build even if the cite process failed. _I also wouldn't recommend doing this_ because [github actions will show a ✅ even if it failed](https://stackoverflow.com/questions/62045967/is-there-a-way-to-continue-on-error-while-still-getting-correct-feedback), so you wouldn't know there was a problem unless you checked the logs.

---

Instead, I would recommend just removing the `default: "[no id info]"` fallback inside `<span class="citation-id">` in `/_includes/citation.html`, and just don't put any `id` in your `sources.yaml`. If you want to show some sort of custom id like "ohbm2024i", you could just put that in the `description`.
